templates/cmake.mpd

```diff
@@ -0,0 +1,158 @@
+cmake_minimum_required(VERSION <%cmake_minimum_required(3.12.0)%>)
+
+<%marker(top)%>
+project(<%project_name%> <%language%>)
+option(BUILD_SHARED_LIBS "Build using shared libraries" <%if(need_staticflags)%>OFF<%else%>ON<%endif%>)
+
+<%if(pre_find_package)%>
+<%pre_find_package%>
+<%endif%>
+<%foreach(packages)%>
+find_package(<%package%> REQUIRED)
+<%endfor%>
+
+set(SOURCE_FILES <%source_files%>)
+<%if(libs || lit_libs || pure_libs)%>
+set(TARGET_LINK_LIBRARIES<%if(libs)%> <%foreach(configurations)%><%fornotfirst("                          ")%>$<$<CONFIG:<%configuration%>>:<%foreach(libs)%><%fornotfirst(" ")%><%lib%>${LIBRARY_DECORATOR}<%lib_modifier%><%endfor%>><%fornotlast("\n")%><%endfor%><%endif%><%if(lit_libs)%> <%lit_libs%><%endif%><%if(pure_libs)%> <%pure_libs%><%endif%>)
```

Comment:
Hmm... I guess I don't fully understand how CMake handles configurations.  Do you have a suggestion on how to make that work?

Reply:
Maybe defaulting to just to the library name if `CONFIG` is empty? It's what I manually changed it to work, but I guess I could have also set the config/build type. I might have to research this more because honestly I probably don't know much more about configurations than you do.

Reply:
I checked in a change to address this.  Give it a try when you get a chance, please.

Reply:
Yes, I think this looks good.